backend/tests/unit/components/permissionHelpers.spec.js

```diff
@@ -0,0 +1,121 @@
+  });
+});
+
+describe('checkWebAdePostPermissions', () => {
+  it('should return no error when user has permission for acrony and acronym is allowed', async () => {
+    // ConfigForm is for WORG, and token has a WORG scope, acronym WORG has true for webade
+    const result = permissionHelpers.checkWebAdePostPermissions(sampleToken, configForm, acronymDetail);
+    expect(result).toBeUndefined();
+  });
+
+  it('should return no error when user has both special scopes', async () => {
+    // ConfigForm is for WORG, and token has a WORG scope, acronym WORG has true for webade
+    const tokenCopy = JSON.parse(JSON.stringify(sampleToken));
+    tokenCopy.realm_access.roles.push(...['WEBADE_PERMISSION', 'WEBADE_PERMISSION_NROS_DMS']);
+    const result = permissionHelpers.checkWebAdePostPermissions(tokenCopy, configForm, acronymDetail, ['WEBADE_PERMISSION', 'WEBADE_PERMISSION_NROS_DMS']);
+    expect(result).toBeUndefined();
+  });
+
+  it('should return an error when user does not have permission for acronym', async () => {
+    // ConfigForm is for ABCD, and token has no scope for that
+    const cfgCopy = JSON.parse(JSON.stringify(configForm));
+    cfgCopy.applicationAcronym = 'ABCD';
+    const result = permissionHelpers.checkWebAdePostPermissions(sampleToken, cfgCopy, acronymDetail);
+    expect(result).toEqual('User lacks permission for \'ABCD\' acronym');
+  });
+
+  it('should return an error when user has no acronym permissions', async () => {
+    // ConfigForm is for ABCD, and token has no scope for that
+    const tokenCopy = JSON.parse(JSON.stringify(sampleToken));
+    tokenCopy.realm_access.roles = [];
+    const result = permissionHelpers.checkWebAdePostPermissions(tokenCopy, configForm, acronymDetail);
+    expect(result).toEqual('User lacks permission for \'WORG\' acronym');
+  });
+
+  it('should return no error when user lacks a desired scope', async () => {
+    // Token is missing WEBADE_PERMISSION_NROS_DMS, and that is required for this operation
+    const tokenCopy = JSON.parse(JSON.stringify(sampleToken));
+    tokenCopy.realm_access.roles.push('WEBADE_PERMISSION');
+    const result = permissionHelpers.checkWebAdePostPermissions(tokenCopy, configForm, acronymDetail, ['WEBADE_PERMISSION', 'WEBADE_PERMISSION_NROS_DMS']);
+    expect(result).toEqual('User is not permitted to submit WebADE config, missing role WEBADE_PERMISSION_NROS_DMS');
+  });
+
+  it('should return an error when acronym is not allowed to use webADE', async () => {
+    // Set acronym detail to not allow webade
+    const adCopy = JSON.parse(JSON.stringify(acronymDetail));
+    adCopy.permissionWebade = false;
+    const result = permissionHelpers.checkWebAdePostPermissions(sampleToken, configForm, adCopy);
+    expect(result).toEqual('Acronym \'WORG\' is not permitted to submit WebADE configs');
+  });
+
+  it('should return an error when acronym is not allowed to do special NROS DMS', async () => {
```

Comment:
Just as a note - the async declaration to the anonymous function is probably not needed in many of these as that there are no explicit awaits done within the body. We can let this code run synchronously (which lets Jest better schedule and manage its threads)

Reply:
Right, copy pasted from one that has an await... Fixed this and will audit the rest to fix during my next PR.